apps/couchbase.go

```diff
@@ -0,0 +1,179 @@
+type LoggingProcessorCouchbaseHTTPAccess struct {
+	confgenerator.ConfigComponent           `yaml:",inline"`
+	confgenerator.LoggingReceiverFilesMixin `yaml:",inline" validate:"structonly"`
+}
+
+// Type returns the string for the couchbase http access logs
+func (lp LoggingProcessorCouchbaseHTTPAccess) Type() string {
+	return "couchbase_http_access"
+}
+
+// Components returns the fluentbit components for the http access logs of couchbase
+func (lp LoggingProcessorCouchbaseHTTPAccess) Components(tag string) []fluentbit.Component {
+	if len(lp.IncludePaths) == 0 {
+		lp.IncludePaths = []string{
+			"/opt/couchbase/var/lib/couchbase/logs/http_access.log",
+			"/opt/couchbase/var/lib/couchbase/logs/http_access_internal.log",
+		}
+	}
+	c := lp.LoggingReceiverFilesMixin.Components(tag)
+	c = append(c,
+		confgenerator.LoggingProcessorParseRegex{
+			Regex: `^(?<client_ip>[^ ]*) [^ ]* (?<user>[^ ]*) \[(?<timestamp>[^\]]*)\] "(?<method>\S+)(?: +(?<path>[^ ]*) +\S*)?" (?<status_code>[^ ]*) (?<response_size>[^ ]*) - (?<message>.*)$`,
+			ParserShared: confgenerator.ParserShared{
+				TimeKey:    "timestamp",
+				TimeFormat: `%d/%b/%Y:%H:%M:%S %z`,
+				Types: map[string]string{
+					"size": "integer",
+					"code": "integer",
+				},
+			},
+		}.Components(tag, "couchbase_http_access")...,
+	)
+	return c
+}
+
+// LoggingProcessorCouchbaseGOXDCR is a struct that iwll generate the fluentbit components for the goxdcr logs
+type LoggingProcessorCouchbaseGOXDCR struct {
+	confgenerator.ConfigComponent           `yaml:",inline"`
+	confgenerator.LoggingReceiverFilesMixin `yaml:",inline" validate:"structonly"`
+}
+
+// Type returns the type string for the cross datacenter logs of couchbase
+func (lg LoggingProcessorCouchbaseGOXDCR) Type() string {
+	return "couchbase_xdcr"
+}
+
+// Components returns the fluentbit components for the couchbase goxdcr logs
+func (lg LoggingProcessorCouchbaseGOXDCR) Components(tag string) []fluentbit.Component {
+	if len(lg.IncludePaths) == 0 {
+		lg.IncludePaths = []string{
+			"/opt/couchbase/var/lib/couchbase/logs/goxdcr.log",
+		}
+	}
+
+	lg.MultilineRules = []confgenerator.MultilineRule{
+		{
+			StateName: "start_state",
+			NextState: "cont",
+			Regex:     `^\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}`,
+		},
+		{
+			StateName: "cont",
+			NextState: "cont",
+			Regex:     `^(?!\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2})`,
+		},
+	}
+	c := lg.LoggingReceiverFilesMixin.Components(tag)
+	c = append(c, confgenerator.LoggingProcessorParseMultilineRegex{
+		LoggingProcessorParseRegexComplex: confgenerator.LoggingProcessorParseRegexComplex{
+			Parsers: []confgenerator.RegexParser{
+				{
+					Regex: `^(?<timestamp>\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}.\d*Z) (?<severity>\w+) (?<log_type>\w+.\w+): (?<message>.*)$`,
+					Parser: confgenerator.ParserShared{
+						TimeKey:    "timestamp",
+						TimeFormat: "%Y-%m-%dT%H:%M:%S.%L",
+					},
+				},
+			},
+		},
+	}.Components(tag, "couchbase_xdcr")...)
+	c = append(c,
```

Comment:
See previous comment. We should use the modiy processor and add the instrumentation source in the same processor.